app/app.ts

```diff
@@ -86,10 +93,40 @@ export class DoseAmigosApp {
                 // schedule an initial token refresh for when the
                 // token expires
                 this.auth.startupTokenRefresh();
+
+                // Notifications are scheduled/rescheduled every 5 minutes
+                this.nIntervId = setInterval(this.scheduleNotifications, 300000);
+
             }
         );
     }
 
+    private scheduleNotifications() {
```

Comment:
This function should probably be in the notification service or a new scheduling service, not here.